File: ccif/src/java/other/org/himalay/ccif/RDC_report.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:04 EST 2016")

public  class RDC_report extends BinStruct implements PublicBinMsg{




    // members variables
    // RDC_centerFreq
    public int RDC_centerFreq ;
    // RDC_transmitterPowerLevel
    public short RDC_transmitterPowerLevel ;
    // bf1
    public BitField_8 bf1 ;

        public RDC_report () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize RDC_centerFreq
        
        // Initialize RDC_transmitterPowerLevel
        
        // Initialize bf1
        bf1 = new BitField_8();
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read RDC_centerFreq
        {RDC_centerFreq=istream.readUnsignedShort(); retVal+=2;}
        // read RDC_transmitterPowerLevel
        {RDC_transmitterPowerLevel=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read bf1
        retVal += bf1.read(istream);

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
        
    
                // write RDC_centerFreq
        ostream.writeShort(RDC_centerFreq); retVal +=2;
        // write RDC_transmitterPowerLevel
        ostream.writeByte(RDC_transmitterPowerLevel); retVal +=1;
        // write bf1
        ostream.writeByte(bf1.getValue()); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("RDC_report\n");
    dc.increaseIndent();
        int retVal= 0;
        // write RDC_centerFreq
        dc.indent();dc.getPs().println("RDC_centerFreq="+RDC_centerFreq+"(0x"+ Integer.toHexString(RDC_centerFreq) + ")" );
        // write RDC_transmitterPowerLevel
        dc.indent();dc.getPs().println("RDC_transmitterPowerLevel="+RDC_transmitterPowerLevel+"(0x"+ Integer.toHexString(RDC_transmitterPowerLevel) + ")" );
        // write bf1
        {dc.indent();dc.getPs().print("reserved: ");dc.getPs().println(BitField_8.toDisplayString(getReserved(),6));dc.indent();dc.getPs().print("RDC_dataRate: ");dc.getPs().println(BitField_8.toDisplayString(getRDC_dataRate(),2));}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for RDC_centerFreq
    //public int getRDC_centerFreq()
    //{
    //    return RDC_centerFreq ;
    //}

    
    // Setter for RDC_centerFreq
    //public void setRDC_centerFreq(int val)
    //{
    //    this.RDC_centerFreq= val;
    //}
    // Getter for RDC_transmitterPowerLevel
    //public short getRDC_transmitterPowerLevel()
    //{
    //    return RDC_transmitterPowerLevel ;
    //}

    
    // Setter for RDC_transmitterPowerLevel
    //public void setRDC_transmitterPowerLevel(short val)
    //{
    //    this.RDC_transmitterPowerLevel= val;
    //}
    // Getter for bf1
    //public BitField_8 getBf1()
    //{
    //    return bf1 ;
    //}

    
    // Setter for bf1
    //public void setBf1(BitField_8 val)
    //{
    //    this.bf1= val;
    //}


    public int getReserved( )
    {
        return ( bf1.getValue() & 0x000000fc ) >> 2 ;
    }
    
    
    public void setReserved(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xffffff03) | ( (val << 2 ) & 0x000000fc));
    }
    
    
    public int getRDC_dataRate( )
    {
        return ( bf1.getValue() & 0x00000003 ) >> 0 ;
    }
    
    
    public void setRDC_dataRate(int val)
    {
        bf1.setValue ( (bf1.getValue() & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/RDC_report.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:16 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "RDC_report.h"

#ifdef org_himalay_ccif_RDC_report__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
RDC_report::RDC_report() {
    references= 0;


    // RDC_centerFreq
    RDC_centerFreq= (ui16)0;
    // RDC_transmitterPowerLevel
    RDC_transmitterPowerLevel= (ui8)0;
    // bf1
    bf1 = 0;
}
// Destructor
RDC_report::~RDC_report() {
    // RDC_centerFreq

    // RDC_transmitterPowerLevel

    // bf1

}

int RDC_report::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // RDC_centerFreq
    {
        RDC_centerFreq=istream.readUI16();
        retVal+=2;
    }
    // RDC_transmitterPowerLevel
    {
        RDC_transmitterPowerLevel=(istream.readUI8());
        retVal+=1;
    }
    // bf1
    bf1= istream.readUI8();
    retVal += 1;
    return retVal;
}

int RDC_report::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read RDC_centerFreq
    {
        RDC_centerFreq=istream.readUI16();
        retVal+=2;
    }
    // read RDC_transmitterPowerLevel
    {
        RDC_transmitterPowerLevel=(istream.readUI8());
        retVal+=1;
    }
    // read bf1
    bf1= istream.readUI8();
    retVal += 1;

    return retVal;
}

int RDC_report::write(DataOutputStream& ostream) { // throws IOException




    int retVal= 0;
    // write RDC_centerFreq
    ostream.writeUI16(RDC_centerFreq);
    retVal +=2;
    // write RDC_transmitterPowerLevel
    ostream.writeUI8(RDC_transmitterPowerLevel);
    retVal +=1;
    // write bf1
    ostream.writeUI8(bf1);
    retVal +=1;

    return retVal;
}

int RDC_report::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("RDC_report\n");
    dc.increaseIndent();
    int retVal= 0;
    // write RDC_centerFreq
    dc.indent();
    dc.getPs().print("RDC_centerFreq=");
    dc.getPs().println((long)RDC_centerFreq);
    // write RDC_transmitterPowerLevel
    dc.indent();
    dc.getPs().print("RDC_transmitterPowerLevel=");
    dc.getPs().println((long)RDC_transmitterPowerLevel);
    // write bf1
    {
        dc.indent();
        dc.getPs().print("reserved: ");
        dc.getPs().printlnBin(getReserved(),6);
        dc.indent();
        dc.getPs().print("RDC_dataRate: ");
        dc.getPs().printlnBin(getRDC_dataRate(),2);
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for RDC_centerFreq
//ui16 RDC_report::getRDC_centerFreq()
//{
//return this->RDC_centerFreq;
//}

// Setter for RDC_centerFreq
//void RDC_report::setRDC_centerFreq(ui16 val)
//{
//this->RDC_centerFreq= val;
//}

// Getter for RDC_transmitterPowerLevel
//ui8 RDC_report::getRDC_transmitterPowerLevel()
//{
//return this->RDC_transmitterPowerLevel;
//}

// Setter for RDC_transmitterPowerLevel
//void RDC_report::setRDC_transmitterPowerLevel(ui8 val)
//{
//this->RDC_transmitterPowerLevel= val;
//}

// Getter for bf1
//ui8 RDC_report::getBf1()
//{
//return this->bf1;
//}

// Setter for bf1
//void RDC_report::setBf1(ui8 val)
//{
//this->bf1= val;
//}

int RDC_report::getReserved( ) {
    return ( this->bf1 & 0x000000fc ) >> 2 ;
}

void RDC_report::setReserved(int val) {
    bf1=(  (this->bf1 & 0xffffff03) | ( (val << 2 ) & 0x000000fc));
}

int RDC_report::getRDC_dataRate( ) {
    return ( this->bf1 & 0x00000003 ) >> 0 ;
}

void RDC_report::setRDC_dataRate(int val) {
    bf1=(  (this->bf1 & 0xfffffffc) | ( (val << 0 ) & 0x00000003));
}

int RDC_report::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_RDC_report__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::RDC_report* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::RDC_report* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code